exporter/internal/queue/merged_context.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package queue // import "go.opentelemetry.io/collector/exporter/internal/queue"
+import (
+	"context"
+	"time"
+
+	"go.uber.org/multierr"
+)
+
+type mergedContext struct {
```

Comment:
The scope is:
1. Deadline is max of all.
2. Keep a list of all SpanContext.

We just need these 2 things, not the whole contexts.